crates/namada/src/ledger/native_vp/masp.rs

```diff
@@ -279,66 +299,385 @@ where
         Ok(())
     }
 
+    /// Look up the IBC denomination from a IbcToken.
+    pub fn query_ibc_denom(
+        &self,
+        token: impl AsRef<str>,
+        owner: Option<&Address>,
+    ) -> Result<String> {
+        let hash = match Address::decode(token.as_ref()) {
+            Ok(Address::Internal(InternalAddress::IbcToken(hash))) => {
+                hash.to_string()
+            }
+            _ => return Ok(token.as_ref().to_string()),
```

Comment:
Are we sure that the `token` in this case is really the `denom` that we're looking for?

Reply:
See https://github.com/anoma/namada/pull/3409#discussion_r1642904303 .